source/Calamari.Tests/Terraform/TerraformFixture.cs

```diff
@@ -0,0 +1,358 @@
+using Calamari.Integration.FileSystem;
+using Calamari.Terraform;
+using Calamari.Tests.Fixtures;
+using Calamari.Tests.Helpers;
+using FluentAssertions;
+using NUnit.Framework;
+using Octostache;
+using Newtonsoft.Json.Linq;
+
+namespace Calamari.Tests.Terraform
+{
+    [TestFixture]
+    [RequiresNonFreeBSDPlatform]
+    [RequiresNon32BitWindows]
+    [RequiresNonMac]
+    [RequiresNonMono]
+    [Category(TestCategory.CompatibleOS.Windows)]
+    public class TerraformFixture
+    {
+        private string customTerraformExecutable;
+
+        [OneTimeSetUp]
+        public async Task InstallTools()
+        {
+            var destinationDirectoryName = TestEnvironment.GetTestPath("TerraformCLIPath");
+
+            if (Directory.Exists(destinationDirectoryName))
+            {
+                var path = Directory.EnumerateFiles(destinationDirectoryName).FirstOrDefault();
+                if (path != null)
+                {
+                    customTerraformExecutable = path;
+                    Console.WriteLine($"Using existing terraform located in {customTerraformExecutable}");
+                    return;
+                }
+            }
+
+            Console.WriteLine("Downloading terraform cli...");
+
+            using (var client = new HttpClient())
+            {
+                var json = await client.GetStringAsync("https://checkpoint-api.hashicorp.com/v1/check/terraform");
+
+                var parsedJson = JObject.Parse(json);
+
+                var downloadBaseUrl = parsedJson["current_download_url"].Value<string>();
+                var currentVersion = parsedJson["current_version"].Value<string>();
+                var fileName = $"terraform_{currentVersion}_windows_amd64.zip";
+
+                if (CalamariEnvironment.IsRunningOnNix)
+                {
+                    fileName = $"terraform_{currentVersion}_linux_amd64.zip";
+                }
+                var zipPath = Path.Combine(Path.GetTempPath(), fileName);
+                using (new TemporaryFile(zipPath))
+                {
+                    using (var fileStream = new FileStream(zipPath, FileMode.Create, FileAccess.Write, FileShare.None))
+                    using (var stream = await client.GetStreamAsync($"{downloadBaseUrl}{fileName}"))
+                    {
+                        await stream.CopyToAsync(fileStream);
+                    }
+
+                    ZipFile.ExtractToDirectory(zipPath, destinationDirectoryName);
+                }
+
+                customTerraformExecutable = Directory.EnumerateFiles(destinationDirectoryName).FirstOrDefault();
+                Console.WriteLine($"Downloaded terraform to {customTerraformExecutable}");
+            }
+        }
+
+        [Test]
+        public void ExtraInitParametersAreSet()
+        {
+            ExecuteAndReturnLogOutput<PlanCommand>(_ =>
+                    _.Set(TerraformSpecialVariables.Action.Terraform.AdditionalInitParams, "-upgrade"), "Simple")
```

Comment:
While this `Action<Dictionary>` approach seems to work, it feels like its not immediately obvious to me, whereas just passing a 
```
new VariableDictionary()
                {
                    {TerraformSpecialVariables.Action.Terraform.AdditionalInitParam, "-upgrade"}
                };
```
makes it more clear what im passing in (you can then take that dictionary and add more defaults to it if need be before its passed along to be invoked)